### PR TITLE
Don't print extra space for `debug_struct` with empty name

### LINKED ISSUE #88261: Don’t print a space after an empty struct name with `Formatter::debug_struct`
I wish I can print a structure like an anonymous structure by specifying an empty structure name using `Formatter::debug_struct`, for example:

```rust
struct Item;

impl Debug for Item {
    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
        f.debug_struct("").field("x", &2).finish()
    }
}
```

I expect `format!("{:?}", Item)` to be `"{ x: 2 }"`, instead, I get `" { x: 2 }"`.

### PR DESCRIPTION
**Pull request by clubby789**
Fixes #88261

I'm not sure if we necessarily want to support what is likely a niche use case, but as it's a quick fix I thought it might be worth having this up and ready for discussion.

@rustbot label +T-libs-api -T-libs

### CODE REVIEW

File: library/core/src/fmt/builders.rs [hunk 1]
```diff
@@ -134,7 +137,13 @@ impl<'a, 'b: 'a> DebugStruct<'a, 'b> {
...
                 writer.write_str(",\n")
             } else {
-                let prefix = if self.has_fields { ", " } else { " { " };
+                let prefix = if self.has_fields {
+                    ", "
+                } else if !self.has_name {
+                    "{ "
+                } else {
+                    " { "
+                };
```
Comment (workingjubilee) [inline library/core/src/fmt/builders.rs:146]: I don't think it's worth this amount of code complication. It's a helper for making your formatter print structs the way the stdlib's impls print structs.

**Review by workingjubilee — COMMENTED** [review]
I'd rather just add anon structs to the language and give them a Debug formatting.

**Comment by workingjubilee** [comment]
Despite my misgivings, if there's some way of addressing the concern without the new field and an additional branch to adjust spacing in every spot, then I think that's cool. Otherwise I genuinely do not think we should penalize most `debug_struct` users.

@rustbot author
r? @workingjubilee

**Comment by JohnCSimon** [comment]
@clubby789 
ping from triage - can you post your status on this PR? This PR has not received an update in a few months. Thank you!

**Comment by clubby789** [comment]
I agree with the comment above, I actually just forgot to close it when it was made 😅 Going to close this now unless I can come up with a good reason/way to do this.

**Pull request closed without merging**